source/SpinalHDL/Data types/bool.rst

```diff
@@ -9,7 +9,52 @@ Bool
 Description
 ^^^^^^^^^^^
 
-The ``Bool`` type corresponds to a boolean value (True or False).
+The ``Bool`` type corresponds to a boolean value (True or False) or a single bit/wire
+used in a hardware design.  While named similarly it should not be confused with
+Scala `Boolean` type which does not describe hardware but truth values in the Scala
+generator code.
+
+An important concept and rule-of-thumb to understand is that the Scala
+`Boolean` type is used in places where ahead-of-time HDL code-generation
```

Comment:
```suggestion
`Boolean` type is used in places where elaboration-time HDL code-generation
```
The term "elaboration" is used in parts of the documentation and should ring a bell for people coming from other HW langs.

Reply:
IDD, maybe the dual but similar meaning use of the term elaboration both
 * in SpinalHDL and 
 * also the next stage of the flow with Verilog/VHDL to achieve synthesis
Is the only thing not clear with this for a first time reader.

In online documentation maybe this can be a linked glossary term with hover-over which defines the term, to clarify the distinction to the first time reader and also provide reference to disambiguate.
This is more a matter for setting up such a glossary of terms/definitions to help here.